Homework-6/Controllers/StudentController.cs

```diff
@@ -0,0 +1,160 @@
+        /// <param name="studentService">The student service.</param>
+        /// <param name="logger">The logger.</param>
+        public StudentController(StudentService studentService, ILogger<StudentController> logger)
+        {
+            _studentService = studentService;
+            _logger = logger;
+        }
+
+        /// <summary>
+        /// Adds a new student.
+        /// </summary>
+        /// <param name="student">The student to add.</param>
+        /// <returns>The ID of the added student.</returns>
+        [HttpPost]
+        public IActionResult AddStudent([FromBody] StudentRequestDTO student)
+        {
+            try
+            {
+                _logger.LogInformation("Adding new student: {@student}", student);
+                var id = _studentService.AddStudent(student);
+                _logger.LogInformation("New student added with ID: {id}", id);
+                return Ok(id.ToString());
+            }
+            catch (ArgumentException ex)
+            {
+                _logger.LogError(ex, "Failed to add student: {message}", ex.Message);
+                return BadRequest(ex.Message);
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, "An unexpected error occurred while adding student.");
+                return StatusCode(500, "An unexpected error occurred while processing the request.");
+            }
+        }
+
+        /// <summary>
+        /// Retrieves a student by ID.
+        /// </summary>
+        /// <param name="id">The ID of the student to retrieve.</param>
+        /// <returns>The student with the specified ID.</returns>
+        [HttpGet("{id}")]
+        public IActionResult GetStudentById(int id)
+        {
+            try
+            {
+                _logger.LogInformation("Fetching student with ID: {id}", id);
+                var student = _studentService.GetStudentById(id);
+                if (student == null)
+                {
+                    _logger.LogInformation("Student not found with ID: {id}", id);
+                    return NotFound($"Student not found with ID: {id}");
+                }
+                _logger.LogInformation("Student list returned successfully");
+                return Ok(student);
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, "An unexpected error occurred while fetching student.");
+                return StatusCode(500, "An unexpected error occurred while processing the request.");
+            }
+        }
+
+        /// <summary>
+        /// Retrieves all students.
+        /// </summary>
+        /// <returns>A list of all students.</returns>
+        [HttpGet]
+        public IActionResult GetAllStudents()
+        {
+            try
+            {
+                _logger.LogInformation("Fetching all students");
+                var students = _studentService.GetAllStudents();
+                if (students.Count == 0)
```

Comment:
Perform these validations in the service layer. Controller should have minimum logic.